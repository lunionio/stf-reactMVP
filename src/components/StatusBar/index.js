import React from "react";
import styled from "styled-components/native";
import { StatusBar as NativeStatusBar, Platform } from "react-native";
import isIphoneX from "../../lib/StatusBarHeight";

const MarginWrapper = styled.View`
  height: ${isIphoneX() ? 44 : 20};

  elevation: ${props => {
    if (props.color) return 6;
    return 1;
  }};

  z-index: ${props => {
    if (props.color) return 6;
    return 1;
  }};

  ${props =>
    props.color
      ? `background-color: ${props.color}`
      : `background-color: ${colors.white}`};
`;

function StatusBar({ color, dark }) {
  return (
    <MarginWrapper color={color}>
      {
        <NativeStatusBar
          barStyle={
            dark || !color
              ? Platform.OS === "ios" ? "dark-content" : "light-content"
              : "light-content"
          }
          backgroundColor={color}
          animated
          translucent
        />
      }
    </MarginWrapper>
  );
}

export default StatusBar;
