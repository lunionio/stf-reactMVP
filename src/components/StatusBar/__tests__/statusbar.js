import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import StatusBar from '../index'

describe('Components - StatusBar', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<StatusBar color={'white'} />).toJSON()
    expect(tree).toMatchSnapshot()
  }),
    it('renders correctly', () => {
      const tree = renderer.create(<StatusBar dark />).toJSON()
      expect(tree).toMatchSnapshot()
    })
})
