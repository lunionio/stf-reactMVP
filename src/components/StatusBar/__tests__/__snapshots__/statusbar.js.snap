// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - StatusBar renders correctly 1`] = `
<View
  color="white"
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "elevation": 6,
        "height": 20,
        "zIndex": 6,
      },
      undefined,
    ]
  }
/>
`;

exports[`Components - StatusBar renders correctly 2`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "elevation": 1,
        "height": 20,
        "zIndex": 1,
      },
      undefined,
    ]
  }
/>
`;
